rust/src/mqtt/parser.rs

```diff
@@ -222,94 +222,104 @@ pub fn parse_connect(i: &[u8]) -> IResult<&[u8], MQTTConnectData> {
     ))
 }
 
-pub fn parse_connack(i: &[u8], protocol_version: u8) -> IResult<&[u8], MQTTConnackData> {
-    let (i, topic_name_compression_response) = be_u8(i)?;
-    let (i, return_code) = be_u8(i)?;
-    let (i, properties) = parse_properties(i, protocol_version == 5)?;
-    Ok((
-        i,
-        MQTTConnackData {
-            session_present: (topic_name_compression_response & 1) != 0,
-            return_code,
-            properties,
-        },
-    ))
+pub fn parse_connack(protocol_version: u8) -> impl Fn(&[u8]) -> IResult<&[u8], MQTTConnackData>
```

Comment:
These formatting changes should go into their own commit, right ?

Reply:
Difficult to separate without undoing a lot of stuff, mixed the `pub` removal with the introduction of the new function structure a bit and then the `rustfmt` also did its thing and changed formatting at the end... Cleaning this up will add quite a bit of manual work to do revert parts of the code to an older state just to get the commits right. Just figured that there was better use of my time... unless you're saying it's a must.

Reply:
The important comment here is indeed another one than Rustfmt thing :

> Thinking about this, I bet you can handle the Incomplete in only one place if you create the right intermediate function parse_message calls some intermediate function to do match message_type {
If this intermediate function returns Incomplete, parse_message` transforms it into a plain error

What do you think ?

Reply:
Yes, I was aware of that and am working on a solution :)